os/kernel/log_dump/log_dump.c

```diff
@@ -278,6 +283,14 @@ int log_dump_set(FAR const char *buffer, size_t buflen)
 		is_started_to_save = true;
 	} else if (strncmp(buffer, LOGDUMP_SAVE_STOP, strlen(LOGDUMP_SAVE_STOP) + 1) == 0) {
 		is_started_to_save = false;
+		/* Now we should compress the partially filled uncompressed buffer */
```

Comment:
If we do stop and start again without read, what happened?

Reply:
Then the last comp block will be present with some old compressed log data and log dump will continue filling latest log dump. But if we try to get size of log dump , it will return error as is_started_to_save is true .
when log dump stop is called again, last comp block will be replaced by latest compressed partially filled buffer and return size will be valid.

Basically , do stop and start again, last_comp_block will have some old compressed log data untill stop is called again .During this period we are not supposed to get log dump size.

Reply:
@ritesh55555 
The question is whether the data remains valid or not?
If we call stop > start > stop > size > read > start .... will we get valid data on the call to size and read?
If we call stop > size > start > stop > size > read > start ... will we get valid data on the second call to size?

Reply:
yes , we will get valid data . I also tested the way you said it. It is workin fine